api/src/opentrons/protocol_api/core/engine/instrument.py

```diff
@@ -88,8 +107,53 @@ def aspirate(
             flow_rate=flow_rate,
         )
 
-    def dispense(self, volume: float, rate: float) -> None:
-        raise NotImplementedError("InstrumentCore.dispense not implemented")
+    def dispense(
+        self,
+        location: Location,
+        well_core: Optional[WellCore],
+        volume: float,
+        rate: float,
+        flow_rate: float,
+    ) -> None:
+        """Dispense a volume of liquid (in microliters/uL) using this pipette
+        into the specified location.
+
+        :param volume: The volume of liquid to dispense, in microliters.
+        :type volume: int or float
+
+        :param location: Where to dispense into. The
+                         robot will dispense into the exact specified location.
+
+        :param well_core: Protocol engine based well core implementation.
+        :type well_core: optional WellCore
+
+        :param rate: Not used in engine based dispense.
+                        Part of the abstract class for legacy purposes.
+
+        :param flow_rate: The absolute dispense flow rate.
+                        `rate` * :py:attr:`default_flow_rate.dispense`.
+        :type rate: float
+        """
```

Comment:
Google-style docstrings to follow latest docstring conventions, including those used by other methods in this file

```suggestion
        """Dispense a given volume of liquid into the specified location.

        Args:
            volume: The volume of liquid to dispense, in microliters.
            location: The exact location to dispense to.
            well_core: The well to dispense to, if applicable.
            rate: Not used in this core.
            flow_rate: The flow rate in µL/s to dispense at.
        """
```